cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -128,29 +128,40 @@ var ErrEnsureEnvPreReqBicepCompileFailed = errors.New("")
 // values are unset. This also requires that the Bicep module can be compiled.
 func (p *BicepProvider) EnsureEnv(ctx context.Context) error {
 	modulePath := p.modulePath()
+
+	// for .bicepparam, we first prompt for environment values before calling compiling bicepparam file
+	// which can reference these values
+	if isBicepParamFile(modulePath) {
+		if err := EnsureSubscriptionAndLocation(ctx, p.envManager, p.env, p.prompters, func(_ account.Location) bool {
+			return true
+		}); err != nil {
```

Comment:
Nit: Instead of passing in this noop style function we should consider just checking if a filter has been provided internally and only executing if it not nil.